tests/rptest/tests/pandaproxy_test.py

```diff
@@ -2235,3 +2236,78 @@ def do_coordinator_change(utils: GroupCoordinatorTransferUtils):
         self.logger.debug(
             "Test offset commit: after second coordinator change")
         do_consumer_offset_commit(new_offset=1)
+
+
+class PandaProxyCompressedBatchesTest(PandaProxyEndpoints):
+    topics = [
+        TopicSpec(),
+    ]
+
+    def __init__(self, context):
+        super(PandaProxyCompressedBatchesTest,
+              self).__init__(context,
+                             extra_rp_conf={
+                                 "enable_leader_balancer": False,
+                                 "group_topic_partitions": 1
+                             })
+
+    @cluster(num_nodes=4)
+    def test_fetch(self):
+        producer = RpkProducer(self.test_context,
+                               self.redpanda,
+                               self.topic,
+                               msg_size=1024,
+                               msg_count=1024,
+                               printable=True,
+                               compression_type="snappy")
+        producer.start()
+        producer.wait()
+
+        self.logger.info("Fetching compressed records from partition 0")
+        fetch_result_raw = self._fetch_topic(self.topic, partition=0)
+        assert fetch_result_raw.status_code == requests.codes.ok
+        result = fetch_result_raw.json()
+        self.logger.debug(f"fetch result {json.dumps(result)}")
+        for item in result:
+            assert item["topic"] == self.topic
+            assert len(item["key"]) > producer.approx_key_size
+            assert len(item["value"]) > producer.approx_msg_size
+            assert item["partition"] == 0
+            assert item["offset"] > -1
```

Comment:
Asking because I don't know:  Is there a way to determine that a record is compressed?  Can this test be refactored in some way to:
1. Produce a single compressed message
2. Consume that message using kafka and confirm it's compressed
3. Consume with the rest proxy and compare the results from \#2?

Reply:
> Asking because I don't know: Is there a way to determine that a record is compressed? Can this test be refactored in some way to:
> 
> 1. Produce a single compressed message
> 2. Consume that message using kafka and confirm it's compressed

I'll see if that's possible. Good idea.

> 3. Consume with the rest proxy and compare the results from [Revert "Merge branch 'leader-stepdown-v2' from Denis" #2](https://github.com/redpanda-data/redpanda/pull/2)?

I'm assuming you mean point (2) instead of that PR?

Reply:
> I'm assuming you mean point (2) instead of that PR?
lol yes.. github trying to be 'helpful'